src/main/java/org/generationcp/middleware/service/api/study/StudyService.java

```diff
@@ -83,4 +84,6 @@
 	long countStudies(StudySearchFilter studySearchFilter);
 
 	boolean studyHasGivenDatasetType(Integer studyId, Integer datasetTypeId);
+
+	boolean environmentHasGivenDatasetType(Integer environmentId, DatasetTypeEnum datasetTypeId);
```

Comment:
We should move this to `StudyInstanceService` and rename as `instanceHasGivenDatasetType`